### PR TITLE
Locale impl

### LINKED ISSUE #312: locale service impl
Implement the locale grpc service

### PR DESCRIPTION
**Pull request by matthew-hagemann**
Closes  #312

### CODE REVIEW

File: provd/example.config.yaml [hunk 1]
```diff
@@ -1,2 +1,2 @@
 Paths:
-  Socket: "/run/test.sock"
+  Socket: "/home/matt/test.sock"
```
Comment (d-loose) [inline provd/example.config.yaml:2]: Was that intentional? I was wondering why my local test setup broke and found this :D

**Review by d-loose — APPROVED** [review]
Tested it using `grpcurl` and using the dart client I'm generating/writing now - works without any issues! :+1:
Possible future improvement, as discussed: return a status from `SetLocale` if the user needs to wait for locale generation to finish. For now we can display a generic loading spinner in the UI.

**Comment by matthew-hagemann** [comment]
Perfect! Thanks Dennis. I'll wait for your impl and we can make sure we are happy end to end before we merge :slightly_smiling_face:

File: provd/example.config.yaml (hunk 1 above)
Comment (matthew-hagemann) [reply provd/example.config.yaml:2]: Haha, no, that was not intentional :sweat_smile: I'll revert that back!

Comment (d-loose) [reply provd/example.config.yaml]: I did the [same](https://github.com/canonical/ubuntu-desktop-provision/compare/a28f5bbc95b932698c3ebee26f94bbc457ec48f4..0bbaa6168c5b3ddedcd039e1091755f890cc9dd6) now.. :grimacing:

**Pull request merged**